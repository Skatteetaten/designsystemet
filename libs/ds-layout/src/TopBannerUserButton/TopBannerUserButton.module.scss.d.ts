declare const classNames: {
  readonly usernameButton: 'usernameButton';
  readonly usernameText: 'usernameText';
};
export default classNames;
