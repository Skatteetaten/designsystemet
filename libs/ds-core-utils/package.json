{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.1.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14"
  },
  "scripts": {}
}
