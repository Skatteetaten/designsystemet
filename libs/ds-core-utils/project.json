{
  "name": "ds-core-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-core-utils/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-core-utils",
        "tsConfig": "libs/ds-core-utils/tsconfig.lib.json",
        "project": "libs/ds-core-utils/package.json",
        "entryFile": "libs/ds-core-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-core-utils/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-core-utils/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-core-utils"],
      "options": {
        "jestConfig": "libs/ds-core-utils/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-core-utils/**/*.css",
          "libs/ds-core-utils/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-core-utils"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-core-utils {args.switches}"
      }
    }
  }
}
