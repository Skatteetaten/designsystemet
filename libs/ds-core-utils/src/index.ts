export * from './SkatteetatenLogo/SkatteetatenLogo';
export * from './base-props.types';
export * from './defaults';
export * from './ExternalLayout/ExternalLayout';
export * from './InternalLayout/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
export * from './hooks/hooks';
export * from './formatter/formatter';
