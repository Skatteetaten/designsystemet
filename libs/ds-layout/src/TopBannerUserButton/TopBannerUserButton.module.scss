@use 'libs/ds-core-designtokens/lib/designtokens/breakpoints';

.username {
  &Button {
    max-width: 4.5rem;

    @media (min-width: breakpoints.$breakpoint-xs) {
      max-width: 6.25rem;
    }

    @media (min-width: breakpoints.$breakpoint-s) {
      max-width: 7.5rem;
    }

    @media (min-width: breakpoints.$breakpoint-m) {
      max-width: 20rem;
    }
  }

  &Text {
    width: 100%;
    text-overflow: ellipsis;
  }
}
