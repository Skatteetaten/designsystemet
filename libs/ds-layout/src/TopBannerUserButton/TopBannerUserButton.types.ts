import { MouseEventHandler, Ref } from 'react';

import { BaseProps } from '@skatteetaten/ds-core-utils';

import { User } from '../TopBannerExternal/TopBannerExternal.types';

export const userRoleArr = ['meg', 'andre', 'virksomhet'] as const;

export interface TopBannerUserButtonProps extends BaseProps {
  ref?: Ref<HTMLButtonElement>;
  /** Hvilken rolle innlogget bruker har. Hvis user.role er lik 'meg' settes automatisk teksten 'Meg selv' ellers settes teksten til user.name */
  user: User;
  /** Callback når knappen trykkes på. */
  onClick?: MouseEventHandler<HTMLButtonElement>;
}
