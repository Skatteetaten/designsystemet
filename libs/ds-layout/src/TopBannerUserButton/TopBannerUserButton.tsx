import { JSX } from 'react';
import { useTranslation } from 'react-i18next';

import { dsI18n, getCommonClassNameDefault } from '@skatteetaten/ds-core-utils';
import {
  AccountMultipleSVGpath,
  BriefcaseSVGpath,
  PersonSVGpath,
} from '@skatteetaten/ds-icons';

import { TopBannerUserButtonProps } from './TopBannerUserButton.types';
import { TopBannerButton } from '../TopBannerButton/TopBannerButton';

import styles from './TopBannerUserButton.module.scss';

export const TopBannerUserButton = ({
  ref,
  id,
  className = getCommonClassNameDefault(),
  lang,
  'data-testid': dataTestId,
  onClick,
  user,
}: TopBannerUserButtonProps): JSX.Element => {
  const { t } = useTranslation('ds_layout', { i18n: dsI18n });

  const buttonText =
    user.role === 'meg' ? t('topbannerbutton.Myself') : user.name;

  let roleSVGpath = PersonSVGpath;
  let title = '';
  if (user.role === 'meg') {
    roleSVGpath = PersonSVGpath;
    title = t('topbannerbutton.MyselfTitle');
  } else if (user.role === 'virksomhet') {
    roleSVGpath = BriefcaseSVGpath;
    title = t('topbannerbutton.CompanyTitle');
  } else if (user.role === 'andre') {
    roleSVGpath = AccountMultipleSVGpath;
    title = t('topbannerbutton.OtherTitle');
  }

  return (
    <TopBannerButton
      ref={ref}
      id={id}
      className={className}
      classNames={
        user.role === 'meg'
          ? undefined
          : {
              container: styles.usernameButton,
              text: styles.usernameText,
            }
      }
      lang={lang}
      data-testid={dataTestId}
      variant={'filled'}
      svgPath={roleSVGpath}
      title={title}
      onClick={onClick}
    >
      {buttonText}
    </TopBannerButton>
  );
};

TopBannerUserButton.displayName = 'TopBannerUserButton';
