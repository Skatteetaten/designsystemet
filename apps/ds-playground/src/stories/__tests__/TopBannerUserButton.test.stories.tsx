import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { dsI18n } from '@skatteetaten/ds-core-utils';

// eslint-disable-next-line @nx/enforce-module-boundaries
import { TopBannerUserButton } from '../../../../../libs/ds-layout/src/TopBannerUserButton/TopBannerUserButton';
// eslint-disable-next-line @nx/enforce-module-boundaries
import {
  TopBannerUserButtonProps,
  userRoleArr,
} from '../../../../../libs/ds-layout/src/TopBannerUserButton/TopBannerUserButton.types';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    await expect(button).toBeInTheDocument();
    await expect(button).toHaveAttribute(attribute, expectedValue);
  };

const meta = {
  component: TopBannerUserButton,
  title: 'Tester/TopBanner/TopBannerUserButton (intern)',
  argTypes: {
    // Baseprops
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    user: { table: { disable: true } },
    // Events
    onClick: { table: { disable: true } },
  },
} as Meta<typeof TopBannerUserButton>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: TopBannerUserButtonProps = {
  user: { role: 'meg' },
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('id', 'dummyIdForwardedFromRef'),
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    await expect(button).toHaveAttribute('id', 'htmlid');
    await expect(button).toHaveClass('dummyClassname');
    await expect(button).toHaveAttribute('lang', 'nb');
    await expect(button).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (Username A2)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    user: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const usernameButtonText = canvas.getByText(
      dsI18n.t('ds_layout:topbannerbutton.Myself')
    );
    const title = canvas.getByText(
      dsI18n.t('ds_layout:topbannerbutton.MyselfTitle')
    );
    await expect(usernameButtonText).toBeInTheDocument();
    await expect(title).toBeInTheDocument();
  },
} satisfies Story;

const TemplateWithAllRoles: StoryFn<typeof TopBannerUserButton> = (args) => (
  <div className={'topbannerButtonContainer'}>
    {userRoleArr.map((role, index) => {
      const name =
        role === 'andre'
          ? 'SØRUMSAND-SCHOENBERG WILLIAM'
          : 'Navn på virksomhet';
      return (
        <TopBannerUserButton
          key={`user_${index}`}
          className={'bottomSpacingXL'}
          {...args}
          user={{ role, name }}
        />
      );
    })}
  </div>
);

export const WithAllRoles = {
  render: TemplateWithAllRoles,
  name: 'With All Roles (Username A2, A3, A6, A7)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    user: {
      table: { disable: false },
      control: { disable: true },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const titleSelf = canvas.getByText(
      dsI18n.t('ds_layout:topbannerbutton.MyselfTitle')
    );
    const titleWork = canvas.getByText(
      dsI18n.t('ds_layout:topbannerbutton.CompanyTitle')
    );
    const titleOther = canvas.getByText(
      dsI18n.t('ds_layout:topbannerbutton.OtherTitle')
    );
    await expect(titleSelf).toBeInTheDocument();
    await expect(titleWork).toBeInTheDocument();
    await expect(titleOther).toBeInTheDocument();

    await expect(
      canvas.getByText(dsI18n.t('ds_layout:topbannerbutton.Myself'))
    ).toBeInTheDocument();
    await expect(
      canvas.getByText('SØRUMSAND-SCHOENBERG WILLIAM')
    ).toBeInTheDocument();
    await expect(canvas.getByText('Navn på virksomhet')).toBeInTheDocument();
  },
} satisfies Story;

export const WithMobileScreen = {
  render: TemplateWithAllRoles,
  name: 'With Small Screen (Username A2, A5)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithBreakpointXS = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-xs (Username A2, A5)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithBreakpointS = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-s (Username A2, A5)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-s',
    },
  },
} satisfies Story;

export const WithBreakpointM = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-m (Username A2, A5)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-m',
    },
  },
} satisfies Story;
