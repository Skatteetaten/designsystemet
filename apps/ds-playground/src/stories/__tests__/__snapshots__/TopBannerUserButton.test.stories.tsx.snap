// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/TopBanner/TopBannerUserButton (intern) Defaults play-test 1`] = `
<div data-test-block="true">
  <button class="_button_1hyk0_1 _button_filled_1hyk0_32"
          type="button"
  >
    <span class="_iconWrapper_1hyk0_18">
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r2:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r2:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
    </span>
    <span class="_buttonText_1hyk0_27">
      Meg selv
    </span>
  </button>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithAllRoles play-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r3:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r3:">
            Innlogget som
          </title>
          <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27">
        Meg selv
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r4:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r4:">
            Innlogget som annen
          </title>
          <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r5:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r5:">
            Innlogget som virksomhet
          </title>
          <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button id="htmlid"
          class="_button_1hyk0_1 _button_filled_1hyk0_32 dummyClassname"
          lang="nb"
          data-testid="123ID"
          type="button"
  >
    <span class="_iconWrapper_1hyk0_18">
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r1:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r1:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
    </span>
    <span class="_buttonText_1hyk0_27">
      Meg selv
    </span>
  </button>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithBreakpointM smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rf:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rf:">
            Innlogget som
          </title>
          <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27">
        Meg selv
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rg:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rg:">
            Innlogget som annen
          </title>
          <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rh:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rh:">
            Innlogget som virksomhet
          </title>
          <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithBreakpointS smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rc:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rc:">
            Innlogget som
          </title>
          <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27">
        Meg selv
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rd:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rd:">
            Innlogget som annen
          </title>
          <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:re:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:re:">
            Innlogget som virksomhet
          </title>
          <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithBreakpointXS smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r9:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r9:">
            Innlogget som
          </title>
          <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27">
        Meg selv
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:ra:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:ra:">
            Innlogget som annen
          </title>
          <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:rb:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:rb:">
            Innlogget som virksomhet
          </title>
          <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithMobileScreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r6:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r6:">
            Innlogget som
          </title>
          <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27">
        Meg selv
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r7:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r7:">
            Innlogget som annen
          </title>
          <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="_button_1hyk0_1 _button_filled_1hyk0_32 bottomSpacingXL _usernameButton_wslqw_1"
            type="button"
    >
      <span class="_iconWrapper_1hyk0_18">
        <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-labelledby="svgtitle-:r8:"
             aria-hidden="false"
             role="img"
        >
          <title id="svgtitle-:r8:">
            Innlogget som virksomhet
          </title>
          <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
          </path>
        </svg>
      </span>
      <span class="_buttonText_1hyk0_27 _usernameText_wslqw_19">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBanner/TopBannerUserButton (intern) WithRef play-test 1`] = `
<div data-test-block="true">
  <button class="_button_1hyk0_1 _button_filled_1hyk0_32"
          type="button"
          id="dummyIdForwardedFromRef"
  >
    <span class="_iconWrapper_1hyk0_18">
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hyk0_18"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r0:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r0:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
    </span>
    <span class="_buttonText_1hyk0_27">
      Meg selv
    </span>
  </button>
</div>
`;
